docker/musl-gcc.sh

```diff
@@ -1,20 +1,82 @@
 #!/bin/bash
 
-# this works around missing libc routines when compiling
-# against a static libstdc++, which we always do on musl after
-# https://github.com/cross-rs/cross/issues/902. the reason
-# otherwise, we are missing crucial routines as as `setlocale`,
-# `__cxa_atexit`, and others.
+# this linker works around missing builtins in older rust versions.
+# we also have custom linker scripts for our static libstdc++ for all versions
+# which is found in `musl-symblink.sh`.
+#
+# for other targets, issues in older versions of compiler-builtins require
+# manually linking to libgcc to compensate for missing builtins.
+# target-specific details include:
+#
+# aarch64-unknown-linux-musl (fixed in 1.48)
+#   https://github.com/rust-lang/compiler-builtins/pull/377
+#
+# armv5te-unknown-linux-musleabi (fixed in 1.65)
+#   missing sync `sync_X_and_fetch`
+#   https://github.com/rust-lang/compiler-builtins/pull/484
+#
+# mips64-unknown-linux-muslabi64, mips64el-unknown-linux-muslabi64  (fixed in 1.65)
+#   missing soft-fp routine `__trunctfsf2`
+#   https://github.com/rust-lang/compiler-builtins/pull/483
 
 set -x
 set -euo pipefail
 
 main() {
-    if [[ $# -eq 0 ]]; then
-        exec "${CROSS_TOOLCHAIN_PREFIX}"gcc "${@}"
+    local minor
+    local flags=()
+    local patched_minor="${CROSS_BUILTINS_PATCHED_MINOR_VERSION:-0}"
+
+    if [[ $# -eq 0 ]] || [[ "${patched_minor}" == 0 ]]; then
```

Comment:
We could probably just remove this wrapper script for just about every target now, actually.